src/rockstor/fs/btrfs.py

```diff
@@ -1549,47 +1552,75 @@ def scrub_status(pool):
     """
     stats = {'status': 'unknown', }
     mnt_pt = mount_root(pool)
+    out3, err3, rc3 = run_command([BTRFS, 'version'])
+    btrfsProgsVers = out3[0].strip().split()[1]
+    # Based on version of btrfs progs, set the offset to parse properly
+    if parse_version(btrfsProgsVers) < parse_version("v5.1.2"):
+        statOffset = 1
+        durOffset = 1
+        fieldOffset = 2
+    else:
+        statOffset = 2
+        durOffset = 3
+        fieldOffset = 4
     out, err, rc = run_command([BTRFS, 'scrub', 'status', '-R', mnt_pt])
     if err != [''] and len(err) > 0:
         if err[0] == "WARNING: failed to read status: Connection reset by " \
                      "peer":
             stats['status'] = 'conn-reset'
             return stats
     if len(out) > 1:
-        if re.search('interrupted', out[1]) is not None:
+        if re.search('interrupted', out[statOffset]) is not None:
             stats['status'] = 'halted'
             # extract the duration from towards the end of the first line eg:
             # "... 2017, interrupted after 00:00:09, not running"
-            dfields = out[1].split()[-3].strip(',').split(':')
+            dfields = out[durOffset].split()[-1].strip(',').split(':')
```

Comment:
@ubenmackin ,

I couldn't test with `btrfs-progs` > 5.1.2 yet, but in 4.19, we need to keep using `.split()[-3]` otherwise we can't catch an interrupted scrub. If [-1] is indeed needed for newer `btrfs-progs`, maybe we could add another version-dependent offset for this specific split?

```suggestion
            dfields = out[durOffset].split()[-haltedDurOffset].strip(',').split(':')
```

As it, the corresponding btrfs unit test fails with:
```
======================================================================
ERROR: test_scrub_status_halted (fs.tests.test_btrfs.BTRFSTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/opt/build/src/rockstor/fs/tests/test_btrfs.py", line 560, in test_scrub_status_halted
    self.assertEqual(scrub_status(pool), expected_results,
  File "/opt/build/src/rockstor/fs/btrfs.py", line 1579, in scrub_status
    (int(dfields[1]) * 60) + int(dfields[2]))
ValueError: invalid literal for int() with base 10: 'running'
```

Reply:
Sorry, not sure how I missed that. I'll get that fixed up.